samples/HelloSignalR/Startup.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            // Add the platform handler to the request pipeline.
+            app.UseIISPlatformHandler();
+
+            if (env.IsDevelopment())
+            {
+                // In case any errors occur.
+                app.UseDeveloperExceptionPage();
+
+                loggerFactory.MinimumLevel = LogLevel.Information;
+                loggerFactory.AddConsole();
+                loggerFactory.AddDebug();
+            }
+
+            var hostname = "http://localhost:5000/";
+
+            // Add token issuing middleware.
+            app.UseOpenIdConnectServer(config =>
+            {
+                // Your Auhentication server.
+                config.Provider = new AuthenticationProvider();
+                config.Issuer = new Uri(hostname);
+            });
+
+            app.UseJwtBearerAuthentication(options =>
+            {
+                // We need this to enable authentication for all requests.
+                options.AutomaticAuthenticate = true;
+
+                // Your audience that has to be allowed by your Authority server.
+                options.Audience = hostname;
+
+                // Authority that issued your token (i.e. your identity server).
+                options.Authority = hostname;
+
+                options.Events = new JwtBearerEvents
+                {
+                    // For SignalR connections default Authorization header does not work,
+                    // Because there are no Headers in WebSockets specification
+                    // This is why we have to implement our own way to retrieve token
+                    // Which can be as easy as adding it to query string
+                    OnReceivingToken = context =>
+                    {
+                        // If the token is not there, look at '?token={token}' query string value
+                        context.Token = context.Token ?? context.Request.Query["access_token"];
+                        return Task.FromResult(true);
+                    }
+                };
+
+                if (env.IsDevelopment())
+                {
+                    // For not requiring HTTPS in localhost.
+                    options.RequireHttpsMetadata = false;
+                }
+            });
+
+            // Add WebSockets handling for SignalR to support it.
+            app.UseWebSockets();
+
+            app.UseSignalR<SimpleConnection>("/signalr");
+
+            app.Use(async (context, next) =>
+             {
+                 var identity = context.User.Identity;
+                 if (!identity.IsAuthenticated)
+                 {
+                     await next();
+                 }
+                 else {
+                     await context.Response.WriteAsync($"Username: {identity.Name}");
+                 }
+             });
+
+            // Configure the HTTP request pipeline for our front-end files.
+            app.UseDefaultFiles();
```

Comment:
I think I will leave it here and add readme.md for project and explain steps how one can use the sample to understand how it works. And then this little middleware starts to make sense.